### PR TITLE
[hubot] Add type for Robot.adapter

### PR DESCRIPTION
**Pull request by markspolakovs**
Hubot adapters are available on the Robot type as `robot.adapter`.
This is used, for example, by Slack, to access the Web API.
This commit adds a type parameter to Robot and the Robot.adapter member.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://slackapi.github.io/hubot-slack/basic_usage#general-web-api-patterns
- [N/A] Increase the version number in the header if appropriate.
- [N/A] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by dirk — APPROVED** [review]
Thanks for improving this; looks good to me! One little suggestion for you about using more generics to avoid an `any` type.

File: types/hubot/index.d.ts
Comment (markspolakovs) [reply types/hubot/index.d.ts]: Oh, I didn't need to know that. That's a good idea, I'll add that to my PR today or tomorrow :)

**Comment by markspolakovs** [comment]
Maintainers: please wait until I resolve the comment above before the merge. (BTW, who should I ping when it's ready? Thanks!)

File: types/hubot/index.d.ts
Comment (paulvanbrenk) [inline types/hubot/index.d.ts]: starting a review, so I get pinged when there's a change.

**Review by paulvanbrenk — CHANGES_REQUESTED** [review]

File: types/hubot/index.d.ts
Comment (paulvanbrenk) [reply types/hubot/index.d.ts]: @markspolakovs when you submit more changes I'll get pinged to review again... which I won't get to before Monday.

**Comment by markspolakovs** [comment]
@dirk how does this look?

**Review by dirk — APPROVED** [review]
@markspolakovs: Looks great; thanks again for putting in the time to contribute this improvement and iterate on it!

@paulvanbrenk: I'm in favor of merging. 🙃

**Pull request merged**
